server/manager/src/main/java/org/apache/accumulo/manager/tableOps/bulkVer2/LoadFiles.java

```diff
@@ -202,104 +212,127 @@ private Client(HostAndPort server, TabletClientService.Client service) {
+      try {
+        var sendTimer = Timer.startNew();
 
-            if (log.isTraceEnabled()) {
-              log.trace("{} asking {} to bulk import {} files for {} tablets", fmtTid, server,
-                  tabletFiles.values().stream().mapToInt(Map::size).sum(), tabletFiles.size());
-            }
+        // Send load messages to tablet servers spinning up work, but do not wait on results.
+        queue.forEach((server, tabletFiles) -> {
 
-            // Tablet servers process tablets serially and perform a single metadata table write for
-            // each tablet. Break the work into per-tablet chunks so it can be sent over multiple
-            // connections to the tserver, allowing each chunk to be run in parallel on the server
-            // side. This allows multiple threads on a single tserver to do metadata writes for this
-            // bulk import.
-            int neededConnections = Math.min(maxConnections, tabletFiles.size());
-            List<Map<TKeyExtent,Map<String,MapFileInfo>>> chunks =
-                new ArrayList<>(neededConnections);
-            for (int i = 0; i < neededConnections; i++) {
-              chunks.add(new HashMap<>());
-            }
-
-            int nextConnection = 0;
-            for (var entry : tabletFiles.entrySet()) {
-              chunks.get(nextConnection++ % chunks.size()).put(entry.getKey(), entry.getValue());
-            }
+          if (log.isTraceEnabled()) {
+            log.trace("{} asking {} to bulk import {} files for {} tablets", fmtTid, server,
+                tabletFiles.values().stream().mapToInt(Map::size).sum(), tabletFiles.size());
+          }
 
-            for (var chunk : chunks) {
-              try {
-                var client = ThriftUtil.getClient(ThriftClientTypes.TABLET_SERVER, server,
-                    manager.getContext(), timeInMillis);
-                // add client to list before calling send in case there is an exception, this makes
-                // sure its returned in the finally
-                clients.add(new Client(server, client));
-                client.send_loadFilesV2(TraceUtil.traceInfo(), manager.getContext().rpcCreds(), tid,
-                    bulkDir.toString(), chunk, setTime);
-              } catch (TException ex) {
-                log.debug("rpc send failed server: {}, {}", server, fmtTid, ex);
-              }
+          // Tablet servers process tablets serially and perform a single metadata table write for
+          // each tablet. Break the work into per-tablet chunks so it can be sent over multiple
+          // connections to the tserver, allowing each chunk to be run in parallel on the server
+          // side. This allows multiple threads on a single tserver to do metadata writes for this
+          // bulk import.
+          int neededConnections = Math.min(maxConnections, tabletFiles.size());
+          if (log.isTraceEnabled()) {
+            if (neededConnections == maxConnections) {
```

Comment:
Should this be `>=`?

Reply:
With the `min` call the equals would do `>=`.  However we probably want `>` and not `>=`, so adjusted it to `>` in b6a132c